server/src/main/java/org/elasticsearch/index/codec/vectors/IVFVectorsWriter.java

```diff
@@ -280,26 +280,27 @@ private void mergeOneFieldIVF(FieldInfo fieldInfo, MergeState mergeState) throws
             IndexInput docs = docsFileName == null ? null : mergeState.segmentInfo.dir.openInput(docsFileName, IOContext.DEFAULT)
         ) {
             final FloatVectorValues floatVectorValues = getFloatVectorValues(fieldInfo, docs, vectors, numVectors);
-            success = false;
-            long centroidOffset;
-            long centroidLength;
+
+            final long centroidOffset;
+            final long centroidLength;
+            final int numCentroids;
+            final int[][] assignmentsByCluster;
+            final float[] calculatedGlobalCentroid = new float[fieldInfo.getVectorDimension()];
             String centroidTempName = null;
-            int numCentroids;
             IndexOutput centroidTemp = null;
-            CentroidAssignments centroidAssignments;
-            float[] calculatedGlobalCentroid = new float[fieldInfo.getVectorDimension()];
+            success = false;
             try {
                 centroidTemp = mergeState.segmentInfo.dir.createTempOutput(mergeState.segmentInfo.name, "civf_", IOContext.DEFAULT);
                 centroidTempName = centroidTemp.getName();
-                centroidAssignments = calculateAndWriteCentroids(
+                CentroidAssignments centroidAssignments = calculateAndWriteCentroids(
```

Comment:
@john-wagster Having the CentroidAssignments here will prevent having the centroids in memory during merge